taskcat/deployer.py

```diff
@@ -0,0 +1,575 @@
+#!/usr/bin/env python
+# -*- coding: UTF-8 -*-
+# author: Santiago Cardenas <sancard@amazon.com>
+from __future__ import print_function
+
+import json
+import argparse
+import os
+import shutil
+import hashlib
+import datetime
+import logging
+import sys
+from collections import OrderedDict
+from .utils import ClientFactory
+from .utils import CFNYAMLHandler
+
+
+class CFNAlchemist(object):
+    OBJECT_REWRITE_MODE = 10
+    BASIC_REWRITE_MODE = 20
+
+    def __init__(
+        self,
+        input_path,
+        target_bucket_name,
+        target_key_prefix=None,
+        output_directory=None,
+        rewrite_mode=OBJECT_REWRITE_MODE,
+        debug=False,
+        dry_run=False
+    ):
+        # create logger
+        self.logger = logging.getLogger('alchemist')
+        self.logger.setLevel(logging.INFO)
+        # create console handler and set level to debug
+        self.ch = logging.StreamHandler()
+        self.ch.setLevel(logging.INFO)
+        # create formatter
+        self.formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+        # add formatter to ch
+        self.ch.setFormatter(self.formatter)
+        # add ch to logger
+        self.logger.addHandler(self.ch)
+
+        # Constants
+        self._UNSUPPORTED_EXT = ['.bz2', '.gz', '.tar', '.zip', '.rar', '.md', '.txt', '.gif', '.jpg', '.png', '.svg', 'jq']
+        self._TEMPLATE_EXT = ['.template', '.json']
+        self._GIT_EXT = ['.git', '.gitmodules', '.gitignore', '.gitattributes']
+        self._EXCLUDED_DIRS = ['.git', 'ci', '.idea', '.vs']
+        self._prod_bucket_name = 'quickstart-reference'
```

Comment:
can we make this configurable? will make CFNAlchemist more useful outside of QuickStarts